httpclient5/src/main/java/org/apache/hc/client5/http/impl/io/BasicHttpClientConnectionManager.java

```diff
@@ -390,22 +391,15 @@ public void release(final ConnectionEndpoint endpoint, final Object state, final
                 return;
             }
             try {
-                if (keepAlive == null) {
-                    this.conn.close(CloseMode.GRACEFUL);
+                if (keepAlive == null && conn != null) {
+                    conn.close(CloseMode.GRACEFUL);
                 }
                 this.updated = System.currentTimeMillis();
-                if (!this.conn.isOpen() && !this.conn.isConsistent()) {
-                    this.route = null;
-                    this.conn = null;
-                    this.expiry = Long.MAX_VALUE;
-                    if (LOG.isDebugEnabled()) {
-                        LOG.debug("{} Connection is not kept alive", id);
-                    }
-                } else {
+                final boolean reusable = conn != null && conn.isOpen() && conn.isConsistent();
```

Comment:
@arturobernalg The variable seems redundant. Looks OK otherwise.

Reply:
changed.